docs/install-midstream.md

```diff
@@ -0,0 +1,148 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  # Replace the version with whatever version you want to install
+  name: serverless-operator-v1-25-0
+  namespace: openshift-marketplace
+spec:
+  displayName: Serverless Operator
+  # Replace the version with whatever version you want to install
+  image: registry.ci.openshift.org/knative/openshift-serverless-v1.25.0:serverless-index
+  publisher: Red Hat
+  sourceType: grpc
+---
+# CatalogSource for main (in development, unstable)
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  # Replace the version with whatever version you want to install
+  name: serverless-operator-nightly
+  namespace: openshift-marketplace
+spec:
+  displayName: Serverless Operator
+  # Replace the version with whatever version you want to install
+  image: registry.ci.openshift.org/knative/openshift-serverless-nightly:serverless-index
+  publisher: Red Hat
+  sourceType: grpc
+```
+
+__CatalogSource images aren't bound to a specific `patch` version, every commit to a specific branch will actually
+change the `vx.y.0` image (they are streams of major versions).__
+
+### Installation
+
+Apply the following subscription referencing the specific catalog source you want to install:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: serverless-operator
+  namespace: openshift-serverless
+spec:
+  channel: stable
+  name: serverless-operator
+  # Change the CatalogSource name
+  source: serverless-operator-v1-24-0
+  sourceNamespace: openshift-marketplace
+  installPlanApproval: Automatic
+```
+
+### Upgrade
+
+Apply the following subscription referencing the specific catalog source you want to upgrade to:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: serverless-operator
+  namespace: openshift-serverless
+spec:
+  channel: stable
+  name: serverless-operator
+  # Change the CatalogSource name
+  source: serverless-operator-v1-25-0
+  sourceNamespace: openshift-marketplace
+  installPlanApproval: Automatic
+```
+
+### Downgrade
+
+OLM **does not** support downgrades, so referencing a previous version in a Subscription isn't supported, so we need a few
+steps to work around the limitation.
+
+1. Delete the `Subscription`
+
+  ```shell
+  oc delete subscriptions.operators.coreos.com -n openshift-serverless serverless-operator
+  ```
+
+2. Delete the `ClusterServiceVersion` associated with the specific release you have installed:
+
+  ```shell
+  oc delete csv -n openshift-serverless serverless-operator.v1.25.0
+  ```
+
+This step deletes the pods in `openshift-serverless`, these aren't critical since they provide a way to install Knative
+components, that means that until step 3 is complete, installation and configuration functionality on KnativeServing,
+KnativeEventing, and KnativeKafka resources won't work but Knative components will be up and running as previously.
```

Comment:
This looks a bit hack-ish but it should be fine for now.
Are we testing this scenario in some way? 
Just to be sure that this doesn't break in the future

Reply:
That's what the OLM team suggested as an escape rule and we're testing it here [1] [while "Continual" tests run](https://github.com/openshift-knative/serverless-operator/blob/f2bd3b46b9774cb0b51afd64f1b57cb3f5b935fd/test/upgrade/upgrade_test.go#L51-L60).

[1] https://github.com/openshift-knative/serverless-operator/blob/f2bd3b46b9774cb0b51afd64f1b57cb3f5b935fd/test/upgrade/installation/serverless.go#L57-L116

Reply:
Ok thanks for the clarification :)